src/test/java/com/google/sps/CommentRetrievalTest.java

```diff
@@ -0,0 +1,99 @@
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+// Using wildcard import since documentation is unclear which package
+// "USES_DEEP_STUBS" is imported from
+
+@RunWith(JUnit4.class)
+public class CommentRetrievalTest {
+
+  private final int HUNDRED = 100;
+  private final int EXCESS_HUNDRED = 200;
+
+  private YouTubeCommentRetriever commentRetriever;
+  // These url's no longer do anything; rather, serve as parameter for the call
+  private final String POPULAR_VIDEO_URL = "dQw4w9WgXcQ";
+  private final String UNPOPULAR_VIDEO_URL = "cA-arJ0T6L4";
+
+  // Simulate a CommentThreadListResponse with exactly as many comments as expected
+  public void setUp(int expectedComments) throws Exception {
+    YouTube mockedYoutube = mock(YouTube.class, RETURNS_DEEP_STUBS);
+    YouTube.CommentThreads.List mockedCommentThreadList =
+        mock(YouTube.CommentThreads.List.class, RETURNS_DEEP_STUBS);
+    when(mockedCommentThreadList.execute())
+        .thenReturn(
+            new CommentThreadListResponse()
+                .setItems(Collections.nCopies(expectedComments, new CommentThread())));
+    when(mockedYoutube.commentThreads().list(anyString())).thenReturn(mockedCommentThreadList);
+    commentRetriever = new YouTubeCommentRetriever(mockedYoutube);
+  }
+
+  // The simplest case: extract 100 comments from a video with more than 100 comments
+  @Test
+  public void testDefaultBehaviour() throws Exception {
+    setUp(HUNDRED);
+    List<CommentThread> comments = commentRetriever.retrieveComments(POPULAR_VIDEO_URL, HUNDRED);
+    Assert.assertEquals(comments.size(), HUNDRED);
+  }
+
+  // Extract more than 100 comments from a video
+  @Test
+  public void testExcessHundredComments() throws Exception {
+    setUp(EXCESS_HUNDRED);
+    List<CommentThread> comments =
+        commentRetriever.retrieveComments(POPULAR_VIDEO_URL, EXCESS_HUNDRED);
+    Assert.assertEquals(comments.size(), EXCESS_HUNDRED);
+  }
+
+  // Only load the comments that are in a video, without crashing
+  @Test
+  public void doesNotAttemptRetrieveExcess() throws Exception {
+    setUp(1);
+    List<CommentThread> comments =
+        commentRetriever.retrieveComments(UNPOPULAR_VIDEO_URL, EXCESS_HUNDRED);
+    Assert.assertEquals(comments.size(), 1);
+  }
+
+  // Retrieve a specific amount of comments less than 100
+  @Test
+  public void retrievesSpecificNumComments() throws Exception {
+    setUp(12);
+    List<CommentThread> comments = commentRetriever.retrieveComments(POPULAR_VIDEO_URL, 12);
+    Assert.assertEquals(comments.size(), 12);
+  }
+
+  // Retrieve a specific amount of comments more than 100
+  @Test
+  public void retrievesSpecificNumCommentsExcessHundred() throws Exception {
+    setUp(120);
+    List<CommentThread> comments = commentRetriever.retrieveComments(POPULAR_VIDEO_URL, 120);
+    Assert.assertEquals(comments.size(), 120);
+  }
+}
```

Comment:
This test is not correct. You never passed in a token, you shouldn't be able to get more than 100 comments.
I would try https://www.javadoc.io/doc/org.mockito/mockito-core/2.8.9/org/mockito/Mockito.html#10 and add responses that would return a token.